### ISSUE TITLE
Copied text lost font-family and font-size style

### ISSUE THREAD

**Issue by Echo-az** [microsoft/roosterjs#1746]
**Describe the bug**
When copy and paste in the same draft, the font-family and font-size style will be lost and fall back to `font-family: -webkit-standard; font-size: medium;`

**To Reproduce**
1. New draft in Outlook iOS/ Outlook Mac
2. Input several text
3. Copy existing text and paste
4. Check the font and font size of pasted text

**Expected behavior**
Pasted(Copied) text should keep the same font style 

**Screenshots**
iOS
<img width="506" alt="image" src="https://user-images.githubusercontent.com/63284518/234227537-1d5e8952-4b3e-4289-bcf5-6f4d04a79e22.png">
macOS demo
<img width="506" alt="image" src="https://user-images.githubusercontent.com/63284518/234230681-a78b309c-e788-4c41-8f46-bb420651361a.png">

**Additional context**
I did some investigation and found it was a regression brought by [commit](https://github.com/microsoft/roosterjs/pull/1496), which changed the tempDiv position. For mac and mobile, the hierarchy is like below:
<img width="687" alt="image" src="https://user-images.githubusercontent.com/63284518/234228620-30eb02ae-504d-4c31-a13a-df6102436e47.png">
There's a `ms-outlook-mac-container` between body and contentDiv(`ms-outlook-mobile-rooster-writer`), after the change in  [commit](https://github.com/microsoft/roosterjs/pull/1496), the tempDiv was no longer the child of `ms-outlook-mac-container` but the child of body. Seems it affected the copied styles.

I'm not very familiar with the copy logic here. Only found some clue above by testing. Could you give any suggestions about how to fix this bug? in roosterjs or in roosterwriter?

**Comment by JiuqingSong** [comment]
Actually you don't need to. Browser will do this for you. You can check out my demo and try copy and paste. I didn't write any code to apply format to copied text but its format is changed when you paste. Browser does this for you when copy.

**Comment by Echo-az** [comment]
So if I do nothing, copied text will be applied the correct format by Browser. But I get the wrong format (mainly fontFamily and fontSize) in OutlookMac and OutlookiOS. The difference between Mac&iOS and RoosterJSDemoSite is the position of "cloneRoot". Do you have any suggestions about how the position of "cloneRoot" affect the applied format?

**Comment by JiuqingSong** [comment]
what is the difference about cloneRoot?

**Comment by Echo-az** [comment]
Seems they are same except for the position.

RoosterJS Demo Site:
<img width="963" alt="image" src="https://github.com/microsoft/roosterjs/assets/63284518/e1b696d7-0199-45b2-9949-3bd8139ca1b5">

Outlook Mac:
<img width="498" alt="image" src="https://github.com/microsoft/roosterjs/assets/63284518/672c29a2-2762-4550-b87c-85b3a795a231">

**Comment by JiuqingSong** [comment]
I'm still not quite understand the problem.

Can you give me a copied HTML string in both cases? For example, with web browser, if you set a font to the temp div, then copy, you will get a copied HTML with the font you have applied. Then in iOS, what did you get? And what do you expect to get?

Thanks.

**Comment by Echo-az** [comment]
**For your question first:**

When I copy in web browser, I got this in my clipboard: 
`<meta charset='utf-8'><span style="color: rgb(0, 0, 0); font-family: Calibri, Arial, Helvetica, sans-serif; font-size: 16px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial; display: inline !important; float: none;">test</span>`
When I copy in OutlookMac, I got this:
`<span style="caret-color: rgb(0, 0, 0); color: rgb(0, 0, 0); font-family: -webkit-standard; font-size: medium; font-style: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(255, 255, 255); text-decoration: none; display: inline !important; float: none;">Test</span>`

I'm expecting to get the same behavior as web browser in OutlookMac, getting the font-family `Calibri...` and font-size `16px`, rather than `-webkit-standard` and `medium`.

------
Actually I didn't set any specific font to it, I got it that setting its css style will make copied content be applied that style. But I'm not talking about setting its css style but changing its position in the hierarchy. 

I found that after this change (https://github.com/microsoft/roosterjs/pull/1496) the copy operation in OutlookMac received different behaviors: before it has the font-family like `Calibri...` and font-size like `16px`, while after it only get default? `-webkit-standard` and `medium` rather than getting its inherited style. The only difference is the position of the temp div.

So I'm wondering how this temp div (cloneRoot) affect the behavior of copy (except that setting its css style will make copied content be applied that style)? Does its position also affect the behavior of copy? Is it ok to move this temp div anywhere in the html hierarchy? Why moving it out of content div affect the behavior of copy?

Tried to explain as much as I can 😂. Sorry and please let me know if anything still unclear.

**Comment by JiuqingSong** [comment]
The copied HTML is generated by browser (or whatever environment where editor locates in). So if browser retrieves styles from its parent containers, changing the position of temp div may cause this result. In our code we create a temp div and put the HTML with the same selection into the temp div, then let browser handle the copy action. To investigate this, you can just create a simple editable DIV, put HTML into it then press Ctrl+C, it should have the same behavior with rooserjs.

I think changing the styles of the temp div should be able to fix the problem you are seeing. But if it does not, maybe you can submit a PR to provide an optional parameter to customize how to create this temp DIV, then you can restore its original behavior.

**Comment by Echo-az** [comment]
Hi Jiuqing, thanks a lot for the details of copy&paste logic. After you said we "put the HTML with the same selection into the temp div", I investigated and found that OutlookMac and RoosterJS Demo didn't put the same html into the temp div. The html of OutlookMac has lost some styles already even before being copied. Then I found that it's because there's some code in rooster-writer to prevent applying font-family and font-size in default format. Any new lines (new html elements) don't have font-family and font-size style, which is applied to the outmost container div. So these styles will lost after being copied, because of being moved out of the container div.
Finally found the root cause. Thank you again for all your help!
